### ISSUE TITLE
Unicode CSV import/export is distorted like ׳“׳•׳‘׳™ ׳׳×׳™׳ ׳•׳§ ׳׳—׳–׳™׳§ ׳‘׳§׳‘׳•׳§ ׳¨׳§׳•׳  in Wordpress 4.9 version

### ISSUE THREAD

**Issue by ofirk** [woocommerce/woocommerce#17914]
## Prerequisites

- [x ] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [x ] The issue still exists against the latest `master` branch of WooCommerce on Github (this is **not** the same version as on WordPress.org!)
- [x ] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1. Once trying to export products with Hebrew language (Unicode) the text is distorted.
2. Once trying to import CSV with Unicode text, the mapping fields page before the import not recognize the text and ignore it.
2. I found it only happens after I updated WordPress to the latest version 4.9. In version 4.8.3 it works like a charm.

## Expected/actual behavior

When I follow those steps, I see for example Unicode text like ׳“׳•׳‘׳™ ׳׳×׳™׳ ׳•׳§ ׳׳—׳–׳™׳§ ׳‘׳§׳‘׳•׳§ ׳¨׳§׳•׳ ׳‘׳¢׳‘׳¨׳™׳×. ׳¦׳‘׳¢׳™׳: ׳•׳¨׳•׳“, ׳×׳›׳׳×. ׳’׳•׳“׳: 40 ׳¡"׳

I was expecting to see Hebrew text.

## Isolating the problem

- [x ] This bug happens with only WooCommerce plugin active
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [x ] I can reproduce this bug consistently using the steps above

## WordPress Environment

<details>
`
### WordPress Environment ###

WC Version: 3.2.5
Log Directory Writable: ✔
WP Version: 4.9
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: he_IL

### Server Environment ###

Server Info: nginx/1.10.1
PHP Version: 7.0.13
PHP Post Max Size: 64 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.29.0
NSS/3.21 Basic ECC

SUHOSIN Installed: –
Max Upload Size: 64 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.2.5
WC Database Prefix: wp_
MaxMind GeoIP Database: ✔
Total Database Size: 33.26MB
Database Data Size: 31.61MB
Database Index Size: 1.65MB
wp_woocommerce_sessions: Data: 0.06MB + Index: 0.00MB
wp_woocommerce_api_keys: Data: 0.00MB + Index: 0.01MB
wp_woocommerce_attribute_taxonomies: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_downloadable_product_permissions: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_order_items: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_tax_rates: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_tax_rate_locations: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_shipping_zones: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_shipping_zone_locations: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_shipping_zone_methods: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_payment_tokens: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_payment_tokenmeta: Data: 0.00MB + Index: 0.00MB
wp_woocommerce_log: Data: 0.00MB + Index: 0.00MB
wp_aryo_activity_log: Data: 0.25MB + Index: 0.00MB
wp_commentmeta: Data: 0.00MB + Index: 0.01MB
wp_comments: Data: 0.01MB + Index: 0.01MB
wp_links: Data: 0.00MB + Index: 0.00MB
wp_newsletter: Data: 0.02MB + Index: 0.00MB
wp_nturl: Data: 0.02MB + Index: 0.00MB
wp_options: Data: 0.81MB + Index: 0.12MB
wp_postmeta: Data: 25.81MB + Index: 1.15MB
wp_posts: Data: 4.44MB + Index: 0.20MB
wp_termmeta: Data: 0.02MB + Index: 0.02MB
wp_terms: Data: 0.00MB + Index: 0.01MB
wp_term_relationships: Data: 0.02MB + Index: 0.04MB
wp_term_taxonomy: Data: 0.00MB + Index: 0.01MB
wp_usermeta: Data: 0.04MB + Index: 0.02MB
wp_users: Data: 0.00MB + Index: 0.01MB
wp_wishlist: Data: 0.02MB + Index: 0.00MB
wp_yoast_seo_links: Data: 0.02MB + Index: 0.02MB
wp_yoast_seo_meta: Data: 0.05MB + Index: 0.00MB

### Templates ###

Overrides: /woocommerce/archive-product.php 
/woocommerce/auth/footer.php
/woocommerce/auth/form-grant-access.php
/woocommerce/auth/form-login.php
/woocommerce/auth/header.php
/woocommerce/cart/cart-empty.php
/woocommerce/cart/cart-item-data.php
/woocommerce/cart/cart-shipping.php 
/woocommerce/cart/cart-totals.php
/woocommerce/cart/cart.php
/woocommerce/cart/cross-sells.php
/woocommerce/cart/mini-cart.php 
/woocommerce/cart/proceed-to-checkout-button.php
/woocommerce/cart/shipping-calculator.php 
/woocommerce/checkout/cart-errors.php
/woocommerce/checkout/form-billing.php
/woocommerce/checkout/form-checkout.php
/woocommerce/checkout/form-coupon.php
/woocommerce/checkout/form-login.php
/woocommerce/checkout/form-pay.php
/woocommerce/checkout/form-shipping.php
/woocommerce/checkout/order-receipt.php
/woocommerce/checkout/payment-method.php
/woocommerce/checkout/payment.php
/woocommerce/checkout/review-order.php
/woocommerce/checkout/terms.php
/woocommerce/checkout/thankyou.php 
/woocommerce/content-product.php
/woocommerce/content-product_cat.php
/woocommerce/content-single-product.php
/woocommerce/content-widget-product.php
/woocommerce/emails/admin-cancelled-order.php
/woocommerce/emails/admin-failed-order.php
/woocommerce/emails/admin-new-order.php
/woocommerce/emails/customer-completed-order.php
/woocommerce/emails/customer-invoice.php
/woocommerce/emails/customer-new-account.php
/woocommerce/emails/customer-note.php
/woocommerce/emails/customer-on-hold-order.php
/woocommerce/emails/customer-processing-order.php
/woocommerce/emails/customer-refunded-order.php
/woocommerce/emails/customer-reset-password.php
/woocommerce/emails/email-addresses.php 
/woocommerce/emails/email-customer-details.php
/woocommerce/emails/email-footer.php
/woocommerce/emails/email-header.php
/woocommerce/emails/email-order-details.php 
/woocommerce/emails/email-order-items.php 
/woocommerce/emails/email-styles.php
/woocommerce/emails/plain/admin-cancelled-order.php
/woocommerce/emails/plain/admin-failed-order.php
/woocommerce/emails/plain/admin-new-order.php
/woocommerce/emails/plain/customer-completed-order.php
/woocommerce/emails/plain/customer-invoice.php
/woocommerce/emails/plain/customer-new-account.php
/woocommerce/emails/plain/customer-note.php
/woocommerce/emails/plain/customer-on-hold-order.php
/woocommerce/emails/plain/customer-processing-order.php
/woocommerce/emails/plain/customer-refunded-order.php
/woocommerce/emails/plain/customer-reset-password.php
/woocommerce/emails/plain/email-addresses.php 
/woocommerce/emails/plain/email-customer-details.php
/woocommerce/emails/plain/email-order-details.php 
/woocommerce/emails/plain/email-order-items.php 
/woocommerce/global/breadcrumb.php
/woocommerce/global/form-login.php
/woocommerce/global/quantity-input.php 
/woocommerce/global/sidebar.php
/woocommerce/global/wrapper-end.php
/woocommerce/global/wrapper-start.php
/woocommerce/loop/add-to-cart.php
/woocommerce/loop/loop-end.php
/woocommerce/loop/loop-start.php
/woocommerce/loop/no-products-found.php
/woocommerce/loop/orderby.php
/woocommerce/loop/pagination.php
/woocommerce/loop/price.php
/woocommerce/loop/rating.php
/woocommerce/loop/result-count.php
/woocommerce/loop/sale-flash.php
/woocommerce/myaccount/dashboard.php
/woocommerce/myaccount/downloads.php 
/woocommerce/myaccount/form-add-payment-method.php
/woocommerce/myaccount/form-edit-account.php
/woocommerce/myaccount/form-edit-address.php
/woocommerce/myaccount/form-login.php 
/woocommerce/myaccount/form-lost-password.php
/woocommerce/myaccount/form-reset-password.php
/woocommerce/myaccount/lost-password-confirmation.php
/woocommerce/myaccount/my-account.php
/woocommerce/myaccount/my-address.php
/woocommerce/myaccount/my-downloads.php
/woocommerce/myaccount/my-orders.php
/woocommerce/myaccount/navigation.php
/woocommerce/myaccount/orders.php 
/woocommerce/myaccount/payment-methods.php
/woocommerce/myaccount/view-order.php
/woocommerce/notices/error.php
/woocommerce/notices/notice.php
/woocommerce/notices/success.php
/woocommerce/order/form-tracking.php 
/woocommerce/order/order-again.php
/woocommerce/order/order-details-customer.php 
/woocommerce/order/order-details-item.php
/woocommerce/order/order-details.php 
/woocommerce/order/tracking.php
/woocommerce/product-searchform.php
/woocommerce/single-product/add-to-cart/external.php
/woocommerce/single-product/add-to-cart/grouped.php
/woocommerce/single-product/add-to-cart/simple.php
/woocommerce/single-product/add-to-cart/variable.php
/woocommerce/single-product/add-to-cart/variation-add-to-cart-button.php
/woocommerce/single-product/add-to-cart/variation.php
/woocommerce/single-product/meta.php
/woocommerce/single-product/photoswipe.php
/woocommerce/single-product/price.php
/woocommerce/single-product/product-attributes.php
/woocommerce/single-product/product-image.php
/woocommerce/single-product/product-thumbnails.php
/woocommerce/single-product/rating.php
/woocommerce/single-product/related.php
/woocommerce/single-product/review-meta.php
/woocommerce/single-product/review-rating.php
/woocommerce/single-product/review.php
/woocommerce/single-product/sale-flash.php
/woocommerce/single-product/share.php
/woocommerce/single-product/short-description.php
/woocommerce/single-product/stock.php
/woocommerce/single-product/tabs/additional-information.php
/woocommerce/single-product/tabs/description.php
/woocommerce/single-product/tabs/tabs.php
/woocommerce/single-product/title.php
/woocommerce/single-product/up-sells.php
/woocommerce/single-product-reviews.php 
/single-product.php 
/woocommerce/taxonomy-product_cat.php
/woocommerce/taxonomy-product_tag.php

`
</details>

**Comment by rodrigoprimo** [comment]
Thanks for your report @ofirk!

> Once trying to export products with Hebrew language (Unicode) the text is distorted.

Are you able to reproduce the same problem if, instead of exporting WooCommerce products, you export WordPress posts or pages? Could you please provide steps that we could follow to reproduce this? I think that a string in Hebrew that we could add to one of the product fields should be enough.

> Once trying to import CSV with Unicode text, the mapping fields page before the import not recognize the text and ignore it.

Could you please provide a sample CSV that we could use to try to reproduce this problem? Could you also, using the sample CSV, give more details about exactly which text is not recognized and ignored?

**Comment by ofirk** [comment]
Hi and hanks @rodrigoprimo for ur reply!

> Are you able to reproduce the same problem if, instead of exporting WooCommerce products, you export WordPress posts or pages? Could you please provide steps that we could follow to reproduce this? I think that a string in Hebrew that we could add to one of the product fields should be enough.

Didn't try to do the same with posts and other stuff, as I only manage a Woo-commerce shop on my server which is limited as I don't have  grant access to whole the platform because it's hosted at my internet marketing provider.
A string u can use is like:

> זוהי שורת ניסיון כדי לוודא את תקינות המערכת, טעינה נסיונית עם מספרים כמו 123 אותיות בעברית וגם סימני פיסוק.

> Could you please provide a sample CSV that we could use to try to reproduce this problem? Could you also, using the sample CSV, give more details about exactly which text is not recognized and ignored?

Please find a sample CSV as follow:
[Sample_CSV.zip](https://github.com/woocommerce/woocommerce/files/1513101/Sample_CSV.zip)

Basically, all cells that had Unicode were ignored. Even cells with Unicode and numbers were ignored.
Cells with text in English and/or digits like "Stock", "Visible" were successfully loaded.

If u need more information, please let me know.
Thanks.

**Comment by claudiosanches** [comment]
@ofirk I made some tests here, but I'm not able to reproduce that issue.
Still there is an input that break the layout because the big name, probably something that should be fixed in the translation on WordPress.org

```
מק\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\ט"
```

This is breaking that CSV, I removed it and important that product without any trouble.

**Comment by mikejolley** [comment]
Fixed in https://github.com/woocommerce/woocommerce/pull/18059
